gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -127,21 +109,158 @@ def test_happy_path_upload_and_get_s3_file(self):
             input_content in response_s3_object
         ), "Stored and retrieved content should match"
 
-    @pytest.mark.skipif(
-        not bucket_name,
-        reason="Skipping this test, since a valid s3 bucket name couldn't be found ",
-    )
     def test_happy_path_delete_s3_file(self):
-        """DELETE a S3 file, then attempt GET to verify it's gone."""
+        """
+        POST an S3 file in a `directoy/filename` format,
+        then, DELETE the S3 file,
+        then attempt GET to verify it's gone.
+        """
+        input_content = "sample_S3_content"
+        self.gen3_workflow.put_bucket_object_with_boto3(
+            content=input_content,
+            object_path=f"{self.s3_storage_config.bucket_name}/{self.s3_folder_name}/{self.s3_file_name}",
+            s3_storage_config=self.s3_storage_config,
+            user=self.valid_user,
+            expected_status=200,
+        )
+
         self.gen3_workflow.delete_bucket_object_with_boto3(
-            object_path=f"{self.bucket_name}/{self.folder_name}/{self.file_name}",
+            object_path=f"{self.s3_storage_config.bucket_name}/{self.s3_folder_name}/{self.s3_file_name}",
+            s3_storage_config=self.s3_storage_config,
             user=self.valid_user,
             expected_status=200,
         )
 
         # We expect the request to return a response with status 404
         self.gen3_workflow.get_bucket_object_with_boto3(
-            object_path=f"{self.bucket_name}/{self.folder_name}/{self.file_name}",
+            object_path=f"{self.s3_storage_config.bucket_name}/{self.s3_folder_name}/{self.s3_file_name}",
+            s3_storage_config=self.s3_storage_config,
             user=self.valid_user,
             expected_status=404,
         )
+
+    ######################## Test /ga4gh/tes/v1/tasks endpoint ########################
+
+    def test_unauthorized_user_cannot_list_tes_tasks(self):
+        """
+        Ensure that an unauthorized user cannot list TES tasks.
+        Expects: HTTP 200 with an empty task list in the response.
+        """
+        response = self.gen3_workflow.list_tes_tasks(
+            user=self.invalid_user,
+            expected_status=200,
+        )
+        assert (
+            "tasks" in response
+            and isinstance(response["tasks"], list)
+            and len(response["tasks"]) == 0
+        ), f"Unauthorized users should receive an empty task list.But found {response} instead"
+
+    def test_unauthorized_user_cannot_create_tes_tasks(self):
+        """
+        Ensure that an unauthorized user cannot create TES tasks.
+        Expects: HTTP 401 Unauthorized response.
+        """
+        # TODO: Work on 401 for user = None, and 403 for user without `create` policy
+        response = self.gen3_workflow.create_tes_task(
+            user=self.invalid_user,
+            expected_status=401,
+        )
+
+    def test_happy_path_create_tes_tasks(self):
```

Comment:
Could we check that the created task completes successfully, and that the outputs are available and correct?